### PR TITLE
Add note about dependency to upgration guide

### LINKED ISSUE #13954: keycloak 19 legacy: unable to deploy war-packaged user storage provider
### Describe the bug

I am about to migrate an existing project (keycloak storage provider) from a wildfly dist to the latest keycloak 19 legacy version. Upgrade to the quarkus dist is not an option yet, since a lot of effort will be needed where we do not have the budget yet.
(we are using EJBs, CDIs and JPA to access a database for user lookup).

We also need to upgrade to get the desparatly needed functionality "update-email" (https://github.com/keycloak/keycloak/pull/7943) working. We have 300K+ users...

The project was running fine on kc15 and also on kc18-legacy though.

During startup following exception is thrown:
```
java.lang.NoClassDefFoundError: Failed to link com/example/keycloak/user/CustomUserStorageProvider (Module "deployment.custom-keycloak-tools.war" from Service Module Loader): org/keycloak/storage/UserStorageProvider
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1090)
	at org.jboss.modules.ModuleClassLoader.doDefineOrLoadClass(ModuleClassLoader.java:351)
	at org.jboss.modules.ModuleClassLoader.defineClass(ModuleClassLoader.java:482)
	at org.jboss.modules.ModuleClassLoader.loadClassLocal(ModuleClassLoader.java:276)
	at org.jboss.modules.ModuleClassLoader$1.loadClassLocal(ModuleClassLoader.java:79)
	at org.jboss.modules.Module.loadModuleClass(Module.java:765)
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:192)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:467)
	at org.jboss.as.ee@26.1.1.Final//org.jboss.as.ee.utils.ClassLoadingUtils.loadClass(ClassLoadingUtils.java:21)
	at org.jboss.as.ee@26.1.1.Final//org.jboss.as.ee.utils.ClassLoadingUtils.loadClass(ClassLoadingUtils.java:14)
	at org.jboss.as.ee@26.1.1.Final//org.jboss.as.ee.component.deployers.InterceptorAnnotationProcessor.processComponentConfig(InterceptorAnnotationProcessor.java:84)
	at org.jboss.as.ee@26.1.1.Final//org.jboss.as.ee.component.deployers.InterceptorAnnotationProcessor.deploy(InterceptorAnnotationProcessor.java:77)
	at org.jboss.as.server@18.1.1.Final//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:182)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1739)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1701)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1559)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at java.base/java.lang.Thread.run(Thread.java:833)

2022-08-24 11:28:27,129 WARN  [org.jboss.modules.define] (MSC service thread 1-2) Failed to define class com.example.keycloak.user.CustomUserStorageProvider in Module "deployment.custom-keycloak-tools.war" from Service Module Loader: java.lang.NoClassDefFoundError: Failed to link com/example/keycloak/user/CustomUserStorageProvider (Module "deployment.custom-keycloak-tools.war" from Service Module Loader): org/keycloak/storage/UserStorageProvider
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1090)
	at org.jboss.modules.ModuleClassLoader.doDefineOrLoadClass(ModuleClassLoader.java:351)
	at org.jboss.modules.ModuleClassLoader.defineClass(ModuleClassLoader.java:482)
	at org.jboss.modules.ModuleClassLoader.loadClassLocal(ModuleClassLoader.java:276)
	at org.jboss.modules.ModuleClassLoader$1.loadClassLocal(ModuleClassLoader.java:79)
	at org.jboss.modules.Module.loadModuleClass(Module.java:765)
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:192)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	at org.jboss.as.ejb3@26.1.1.Final//org.jboss.as.ejb3.deployment.processors.BusinessViewAnnotationProcessor.getEjbClass(BusinessViewAnnotationProcessor.java:258)
	at org.jboss.as.ejb3@26.1.1.Final//org.jboss.as.ejb3.deployment.processors.BusinessViewAnnotationProcessor.deploy(BusinessViewAnnotationProcessor.java:89)
	at org.jboss.as.server@18.1.1.Final//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:182)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1739)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1701)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1559)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at java.base/java.lang.Thread.run(Thread.java:833)

2022-08-24 11:28:27,131 ERROR [org.jboss.msc.service.fail] (MSC service thread 1-2) MSC000001: Failed to start service jboss.deployment.unit."custom-keycloak-tools.war".POST_MODULE: org.jboss.msc.service.StartException in service jboss.deployment.unit."custom-keycloak-tools.war".POST_MODULE: WFLYSRV0153: Failed to process phase POST_MODULE of deployment "custom-keycloak-tools.war"
	at org.jboss.as.server@18.1.1.Final//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:189)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.startService(ServiceControllerImpl.java:1739)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$StartTask.execute(ServiceControllerImpl.java:1701)
	at org.jboss.msc@1.4.13.Final//org.jboss.msc.service.ServiceControllerImpl$ControllerTask.run(ServiceControllerImpl.java:1559)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1990)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads@2.4.0.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NoClassDefFoundError: Failed to link com/example/keycloak/user/CustomUserStorageProvider (Module "deployment.custom-keycloak-tools.war" from Service Module Loader): org/keycloak/storage/UserStorageProvider
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1012)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1090)
	at org.jboss.modules.ModuleClassLoader.doDefineOrLoadClass(ModuleClassLoader.java:351)
	at org.jboss.modules.ModuleClassLoader.defineClass(ModuleClassLoader.java:482)
	at org.jboss.modules.ModuleClassLoader.loadClassLocal(ModuleClassLoader.java:276)
	at org.jboss.modules.ModuleClassLoader$1.loadClassLocal(ModuleClassLoader.java:79)
	at org.jboss.modules.Module.loadModuleClass(Module.java:765)
	at org.jboss.modules.ModuleClassLoader.findClass(ModuleClassLoader.java:192)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClassUnchecked(ConcurrentClassLoader.java:410)
	at org.jboss.modules.ConcurrentClassLoader.performLoadClass(ConcurrentClassLoader.java:398)
	at org.jboss.modules.ConcurrentClassLoader.loadClass(ConcurrentClassLoader.java:116)
	at org.jboss.as.ejb3@26.1.1.Final//org.jboss.as.ejb3.deployment.processors.BusinessViewAnnotationProcessor.getEjbClass(BusinessViewAnnotationProcessor.java:258)
	at org.jboss.as.ejb3@26.1.1.Final//org.jboss.as.ejb3.deployment.processors.BusinessViewAnnotationProcessor.deploy(BusinessViewAnnotationProcessor.java:89)
	at org.jboss.as.server@18.1.1.Final//org.jboss.as.server.deployment.DeploymentUnitPhaseService.start(DeploymentUnitPhaseService.java:182)
	... 8 more
```

But the class file is in the deployments directory, so something must be wrong in the server not picking up the class files.

Attached, to reproduce the issue, there is the pacakged war-file and also the source code (jar).

Sidenote: packaging as jar did not solve the problem. Anyways I need a war to be able to add additional needed libs.

### Version

19.0.1 legacy

### Expected behavior

the user storage provider should be deployed correctly and usable

### Actual behavior

user storage provider cannot be used

### How to Reproduce?

deploy the provided war file (remove the .zip at the end) and start kc-19-legacy
[custom-keycloak-tools.war.zip](https://github.com/keycloak/keycloak/files/9415006/custom-keycloak-tools.war.zip)

here is the source code (again remove the zip extension):
[custom-keycloak-tools-sources.jar.zip](https://github.com/keycloak/keycloak/files/9415016/custom-keycloak-tools-sources.jar.zip)

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by hmlnarik**
Fixes: keycloak/keycloak#13954

### CODE REVIEW

**Comment by hmlnarik** [comment]
None of the test failures relate to this PR

File: upgrading/topics/keycloak/changes-19_0_0.adoc [hunk 1]
```diff
@@ -111,6 +111,21 @@ Contains all REST endpoints that directly operate on the legacy store, and have
 These modules will be available as long as legacy stores will be supported.
 After that period, they will be removed.
 
+This change impacts deployments of existing user storage providers. If your user storage provider is deployed
```
Comment (ahus1) [inline upgrading/topics/keycloak/changes-19_0_0.adoc]: Adding the hint about the Wildfly distribution as users of the Quarkus distribution might not know what this is all about. I prefer "Wildfly distribution" over "legacy distribution", as "legacy" in this chapter usually refers to the "legacy store".

```suggestion
This change impacts deployments of existing user storage providers in the Wildfly distribution. If your user storage provider is deployed
```

File: upgrading/topics/keycloak/changes-19_0_0.adoc [hunk 2]
```diff
@@ -111,6 +111,21 @@ Contains all REST endpoints that directly operate on the legacy store, and have
...
+as a WAR archive, you need to add the `META-INF/jboss-deployment-structure.xml` file into that archive
+stating the modified dependencies, i.e. with the following contents:
```
Comment (ahus1) [inline upgrading/topics/keycloak/changes-19_0_0.adoc]: Latin words and abbreviations like `i.e.` should avoided in the current style guide. 

```suggestion
stating the modified dependencies as shown below:
```

**Review by ahus1 — CHANGES_REQUESTED** [review]
Thank your for adding this missing bit to the guide. I left two small review comments below.

**Review by ahus1 — APPROVED** [review]

**Pull request merged**